Applications/WebPortal/WebApp/library/source/pages/account_directory_page/account_directory_page.tsx

```diff
@@ -1,18 +1,230 @@
+import { css, StyleSheet } from 'aphrodite';
 import * as React from 'react';
 import { DisplaySize } from '../../display_size';
+import { AccountDirectoryModel, FilterBar } from '.';
+import { GroupCard } from './group_card';
 
 interface Properties {
 
   /** Determines the layout used to display the page. */
   displaySize: DisplaySize;
+
+  /** Called when the user wants to make a new group. */
+  onNewGroupClick?: () => {};
+
+  /** Called when the user wants to make a new account. */
+  onNewAccountClick?: () => {};
+
+  /** Model that contains imformation about the accounts. */
+  model: AccountDirectoryModel;
 }
 
-/** Displays an account's profile page. */
-export class AccountDirectoryPage extends React.Component<Properties> {
+interface State {
+  filter: string;
+}
+
+/** Displays an directory of accounts. */
+export class AccountDirectoryPage extends React.Component<Properties, State> {
+  constructor(props: Properties) {
+    super(props);
+    this.state = {
+      filter: ''
+    };
+    this.onChange = this.onChange.bind(this);
+  }
+
   public render(): JSX.Element {
+    const contentWidth = (() => {
+      switch(this.props.displaySize) {
+        case DisplaySize.SMALL:
+          return AccountDirectoryPage.STYLE.contentSmall;
+        case DisplaySize.MEDIUM:
+          return AccountDirectoryPage.STYLE.contentMedium;
+        case DisplaySize.LARGE:
+          return AccountDirectoryPage.STYLE.contentLarge;
+      }
+    })();
+    const headerBoxStyle = (() => {
+      if(this.props.displaySize === DisplaySize.SMALL) {
+        return AccountDirectoryPage.STYLE.verticalHeaderBox;
+      } else {
+        return AccountDirectoryPage.STYLE.horizontalHeaderBox;
+      }
+    })();
+    const buttonBoxStyle = (() => {
+      if(this.props.displaySize === DisplaySize.SMALL) {
+        return AccountDirectoryPage.STYLE.buttonBoxSmall;
+      } else {
+        return AccountDirectoryPage.STYLE.buttonBox;
+      }
+    })();
+    const buttonStyle = (() => {
+      if(this.props.displaySize === DisplaySize.SMALL) {
+        return AccountDirectoryPage.DYNAMIC_STYLE.buttonSmall;
+      } else {
+        return AccountDirectoryPage.DYNAMIC_STYLE.button;
+      }
+    })();
+    const cards = (() => {
+      const cards = [];
+      for (const group of this.props.model.groups) {
+        cards.push(
+          <GroupCard key={group.id}
+            displaySize={this.props.displaySize}
+            group={group}
+            accounts={[]}
```

Comment:
You will have to populate the accounts prop. This is the most challenging aspect of this page. You need to keep track of which GroupCard's have been expanded by the user and then when a user expands a GroupCard you need to load the list of accounts for that group from the model.